packages/overlay/src/integrations/sentry/data/useSentryLogs.tsx

```diff
@@ -0,0 +1,27 @@
+import { useContext } from 'react';
+import useSentryStore from '../store';
+import { isLocalTrace } from '../store/helpers';
+import { SentryEventsContext } from './sentryEventsContext';
+
+export const useSentryLogs = (traceId?: string) => {
+  useContext(SentryEventsContext);
+  const { logsById, logsByTraceId } = useSentryStore(state => ({
+    logsById: state.logsById,
+    logsByTraceId: state.logsByTraceId,
+  }));
+
+  const allLogs = Array.from(traceId ? (logsByTraceId.get(traceId) ?? []) : logsById.values());
+  const localLogs = allLogs.filter(item => item.trace_id && isLocalTrace(item.trace_id));
```

Comment:
shouldn't we memoize these?

```typescript
  const allLogs = useMemo(
    () => Array.from(traceId ? (logsByTraceId.get(traceId) ?? []) : logsById.values()),
    [traceId, logsById, logsByTraceId]
  );
  const localLogs = useMemo(
    () => allLogs.filter(item => item.trace_id && isLocalTrace(item.trace_id)),
    [allLogs]
  );
```

Reply:
I think this might be done downstream? I'm also not very clear on `useMemo()`. Maybe try in a follow up?